projects/frontegg-app/src/lib/frontegg-user-subscription.service.ts

```diff
@@ -0,0 +1,76 @@
+import { Subscription, PartialObserver, BehaviorSubject } from 'rxjs';
+import { AuthState } from '@frontegg/redux-store';
+import { Injectable } from '@angular/core';
+
+import { FronteggState } from '@frontegg/redux-store';
+import { FronteggAppService } from './frontegg-app.service';
+
+export type User = FronteggState['auth']['user'];
+
+/**
+ * A service for managing user state subscription
+ * The service gives the ability to subscribe to user state change and get a manipulated data when the user state changes
+ */
+@Injectable({
+  providedIn: 'root',
+})
+export class FronteggUserSubscriptionService {
+  private userStateSubject = new BehaviorSubject<any>(undefined);
+    
+  constructor(private fronteggAppService: FronteggAppService) {
+    const state = this.fronteggAppService.fronteggApp.store.getState() as FronteggState;
```

Comment:
`as FronteggState`
maybe we can use `getState<FronteggState>()`

Reply:
doesn't work